libkineto/src/CuptiActivityPlatform.cpp

```diff
@@ -0,0 +1,31 @@
+#include <chrono>
+
+namespace chrono = std::chrono;
+
+namespace KINETO_NAMESPACE {
+
+#ifdef _WIN32
+uint64_t epochs_diff() {
+  // On Windows, steady_clock wraps the QueryPerformanceCounter function.
+  // https://docs.microsoft.com/en-us/cpp/standard-library/steady-clock-struct?view=msvc-160
+  auto steady =
+      chrono::time_point_cast<chrono::nanoseconds>(chrono::steady_clock::now());
+  auto system =
+      chrono::time_point_cast<chrono::nanoseconds>(chrono::system_clock::now());
+
+  auto time_since_unix = system.time_since_epoch().count();
+  auto time_since_boot = steady.time_since_epoch().count();
+  return time_since_unix - time_since_boot;
+}
+
+uint64_t unixEpochTimestamp(uint64_t ts) {

```

Comment:
Does it make sense to implement this in libkineto/include/time_since_epoch.h instead?

Reply:
I don't think so. `time_since_epoch.h` is a public header and this function purely serve as a implementation detail to make cupti timestamp on windows have the same semantics as declared in this header. It is not meant to be used by library users.

Also, there is function static variable to avoid computing the diff repeatedly, so there will be an implementation in a cpp file anyway.

Reply:
but I am open minded to merge it with some other cupti related files.

Reply:
OK makes sense. We can leave it here for now - we have some other platform specific stuff elsewhere (ThreadUtil.cpp), so maybe at some point it makes sense to make a PlatformAbstraction or similar.